src/main/java/mate/academy/spring/controller/UserController.java

```diff
@@ -0,0 +1,30 @@
+package mate.academy.spring.controller;
+
+import java.util.NoSuchElementException;
+import mate.academy.spring.mapper.impl.response.UserResponseMapper;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final UserResponseMapper userResponseMapper;
+    private final UserService userService;
+
+    public UserController(UserResponseMapper userResponseMapper, UserService userService) {
+        this.userResponseMapper = userResponseMapper;
+        this.userService = userService;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto getByEmail(@RequestParam String email) {
+        return userResponseMapper.toDto(
+                userService.findByEmail(email).orElseThrow(
+                        () -> new NoSuchElementException("Couldn't find user by email: "

```

Comment:
In my opinion, we should use `DataProcessingException`

Reply:
I'm not sure if exceptions need to be handled at the controller level at all)))